internal/handlers/common/msg_getmore.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn/conninfo"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgGetMore is a common implementation of the `getMore` command.
+func MsgGetMore(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = Unimplemented(document, "comment", "maxTimeMS"); err != nil {
+		return nil, err
+	}
+
+	db, err := GetRequiredParam[string](document, "$db")
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	collection, err := GetRequiredParam[string](document, "collection")
+	if err != nil {
+		return nil, NewCommandErrorMsg(ErrBadValue, `required parameter "collection" is missing`)
+	}
+
+	cursorIDValue, err := document.Get("getMore")
+	if err != nil {
+		return nil, NewCommandErrorMsg(ErrBadValue, `required parameter "getMore" is missing`)
+	}
+
+	var cursorID int64
+	var ok bool
+
+	if cursorID, ok = cursorIDValue.(int64); !ok {
+		return nil, NewCommandErrorMsg(
+			ErrTypeMismatch,
+			fmt.Sprintf(
+				`BSON field 'getMore.getMore' is the wrong type '%s', expected type 'long'`,
+				AliasFromType(cursorIDValue),
+			),
+		)
+	}
+
+	if cursorID != 1 {
```

Comment:
I see, so get more cursorID can only be 1 at the moment :+1: . https://github.com/FerretDB/FerretDB/issues/1811